### PR TITLE
Add form alert atom to cf-forms

### PR DESCRIPTION
**Pull request by jimmynotjim**
There's a need for default and warning states of the existing
error message atom. Converting the error message to a form alert atom
provides a more robust use case and follows our Atomic Design and BEM
principles.

## Additions

- Add form alert atom to cf-forms

## Testing

1. Follow the [testing locally instructions](https://github.com/cfpb/capital-framework/blob/canary/CONTRIBUTING.md#testing-components-locally)
1. Run `gulp build` && `npm start`
1. Navigate to `http://localhost:3000/components/cf-forms/#form-alerts`

## Screenshots

screen shot 2018-07-12 at 2 49 24 pm

## Notes

- I'm entirely open to feedback on the descriptive text after the heading. We didn't have anything existing so it's a best shot at it.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Internet Explorer 8, 9, 10, and 11
- [x] Edge
- [x] iOS Safari
- [ ] Chrome for Android

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [x] Flexible from small to large screens
- [x] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

**Comment by anselmbradford** [comment]
Might this alert ever be used outside of forms? e.g. could it be `a-alert`?

**Comment by jimmynotjim** [comment]
@anselmbradford We discussed that, but I don't really see where it would be used outside a form. The alert adds context to a page's response to a user's input, but we don't have non-form inputs other than the menus. 

With notifications it made more sense to make them form-agnostic, we use them in quite a few places as a "notice" without the user having taken any action other than navigating to the page, but that seems an unlikely scenario with alerts.

File: src/cf-forms/usage.md [hunk 1]
```diff
@@ -746,34 +746,139 @@ have helper text that appears below the main label text.
...
+<div class="o-form_group">
+    <div class="m-form-field m-form-field__success">
+        <label class="a-label__heading" for="form-input-error">Label</label>
+        <input class="a-text-input a-text-input__success"
+               type="text"
+               value="Valid input"
+               id="form-input-error"
+               aria-describedby="form-input-error_message">
+        <div class="a-form-alert a-form-alert__success" role="alert">
+            {% include icons/info-round.svg %}
```
Comment (Scotchester) [inline src/cf-forms/usage.md]: This should be `check-round`.

File: src/cf-forms/usage.md [hunk 2]
```diff
@@ -746,34 +746,139 @@ have helper text that appears below the main label text.
...
+<div class="o-form_group">
+    <div class="m-form-field m-form-field__success">
+        <label class="a-label__heading" for="form-input-error">Label</label>
+        <input class="a-text-input a-text-input__success"
+               type="text"
+               value="Valid input"
+               id="form-input-error"
+               aria-describedby="form-input-error_message">
+        <div class="a-form-alert a-form-alert__success" role="alert">
+            {% raw %}{% include icons/info-round.svg %}{% endraw %}
```
Comment (Scotchester) [inline src/cf-forms/usage.md]: And here.

**Pull request merged**